pygmt/datasets/samples.py

```diff
@@ -1,13 +1,91 @@
 """
 Functions to load sample data.
 """
+import warnings
+
 import pandas as pd
+from pygmt.exceptions import GMTInvalidInput
 from pygmt.src import which
 
 
-def load_japan_quakes():
+def list_sample_data():
+    """
+    Report datasets available for tests and documentation examples.
+
+    Returns
+    -------
+    dict
+        Names and short descriptions of available sample datasets.
+
+    See Also
+    --------
+    load_sample_data : Load an example dataset from the GMT server.
     """
-    Load a table of earthquakes around Japan as a pandas.DataFrame.
+    names = {
+        "japan_quakes": "Table of earthquakes around Japan from NOAA NGDC database",
+        "ocean_ridge_points": "Table of ocean ridge points for the entire world",
+        "bathymetry": "Table of ship bathymetric observations off Baja California",
+        "usgs_quakes": "Table of global earthquakes from the USGS",
+        "fractures": "Table of hypothetical fracture lengths and azimuths",
+        "hotspots": "Table of locations, names, and symbol sizes of hotpots from "
+        " Mueller et al., 1993",
+        "mars_shape": "Table of topographic signature of the hemispheric dichotomy of "
+        " Mars from Smith and Zuber (1996)",
+    }
+    return names
+
+
+def load_sample_data(name):
```

Comment:
I'm thinking if we can merge the `list_sample_data()` function into `load_sample_data()`. Calling `load_sample_data()` without giving a name can return the name-description dict.

Reply:
I'd prefer to keep them separate even though it requires two dictionaries because I think overall it's simpler to have each function have one purpose.